app/models/concerns/user_access_key_overrides.rb

```diff
@@ -5,69 +5,30 @@
 module UserAccessKeyOverrides
   extend ActiveSupport::Concern
 
-  attr_accessor :user_access_key
-
   def valid_password?(password)
-    return false if encrypted_password.blank?
-    begin
-      unlock_user_access_key(password)
-    rescue Pii::EncryptionError => err
-      log_error(err)
-      return false
-    end
-    Devise.secure_compare(encrypted_password, user_access_key.encrypted_password)
-  end
-
-  def unlock_user_access_key(password)
-    self.user_access_key = build_user_access_key(password).unlock(encryption_key)
+    Encryption::PasswordVerifier.verify(
+      password: password,
+      digest: encrypted_password_digest
+    )
   end
 
   def password=(new_password)
     @password = new_password
-    encrypt_password(@password) if @password.present?
-  end
-
-  def authenticatable_salt
-    password_salt
+    return if @password.blank?
+    self.encrypted_password_digest = Encryption::PasswordVerifier.digest(@password)
+    # Until we drop the old columns, still write to them so that we can rollback
+    write_legacy_password_attributes
```

Comment:
I know this is temporary, but it seems a bit awkward and when we do remove the legacy attributes, we'll have to make changes in multiple places.

What about something like:
```
digest = Encryption::PasswordVerifier.digest(@password)
self.encrypted_password_digest = digest.to_s
write_legacy_password_attributes(digest)
```
where `Encryption::PasswordVerifier.digest` returns the `PasswordDigest` instead of the stringified version?

Then we don't need `.parse_from_string` at all, just need to do it when testing probably.